Network/Socket.hs

```diff
@@ -52,16 +52,17 @@
 -- >               , addrSocketType = Stream
 -- >               }
 -- >         head <$> getAddrInfo (Just hints) mhost (Just port)
--- >     open addr = do
--- >         sock <- socket (addrFamily addr) (addrSocketType addr) (addrProtocol addr)
--- >         setSocketOption sock ReuseAddr 1
--- >         withFdSocket sock setCloseOnExecIfNeeded
--- >         bind sock $ addrAddress addr
--- >         listen sock 1024
--- >         return sock
--- >     loop sock = forever $ do
--- >         (conn, _peer) <- accept sock
--- >         void $ forkFinally (server conn) (const $ gracefulClose conn 5000)
+-- >     open addr = E.bracketOnError
+-- >         (socket (addrFamily addr) (addrSocketType addr) (addrProtocol addr))
+-- >         close $ \sock -> do
+-- >             setSocketOption sock ReuseAddr 1
+-- >             withFdSocket sock setCloseOnExecIfNeeded
+-- >             bind sock $ addrAddress addr
+-- >             listen sock 1024
+-- >             return sock
+-- >     loop sock = forever $ E.bracketOnError (accept sock) (close . fst)
+-- >         $ \(conn, _peer) -> void $
+-- >             forkFinally (server conn) (const $ gracefulClose conn 5000)
```

Comment:
I don't think it's especially necessary here, as only `fork` involved this far, but a nice to have in case more setup before `fork` is added in a real case.

Reply:
Please add a comment to describe your intention of this redundancy.